presto-parser/src/main/java/com/facebook/presto/sql/parser/SqlParser.java

```diff
@@ -231,11 +231,7 @@ public void exitNonReserved(SqlBaseParser.NonReservedContext context)
             context.getParent().removeLastChild();
 
             Token token = (Token) context.getChild(0).getPayload();
-            if (token.getText().equalsIgnoreCase("CURRENT_ROLE")) {
-                warningConsumer.accept(new ParsingWarning(format("Reserved word used: %s", token.getText()), token.getLine(), token.getCharPositionInLine()));
-            }
-
-            if (SqlParserOptions.RESERVED_WORDS_WARNING.contains(token.getText())) {
+            if (SqlParserOptions.RESERVED_WORDS_WARNING.contains(token.getText().toUpperCase())) {
```

Comment:
We structure commits logically, which means that this bug fix should belong to the first commit (if you want to keep 2 separate commits). Since adding `CURRENT_ROLE` to `RESERVED_WORDS_WARNING` is very much related and very short you can merge the two commits into one. Please do `git push -f` to update the PR after you restructure your commits. Thanks!

Reply:
thanks for explaining this :) I've squashed all my commits into one with the commit message most relevant to the issue. Happy to make more changes if something seems off